### ISSUE TITLE
Zenfolio moving to HTTPS-only

### ISSUE THREAD

**Issue by chris-zenfolio** [lildude/phpZenfolio#28]
Hello,

Over the past 18 months, Zenfolio has been working to strengthen and improve its infrastructure to provide a more reliable and secure platform for all of our photographers, photographers’ clients, partners, and developers.  While performing these improvements, we are transitioning the Zenfolio API to an HTTPS-only environment.  Due to recent infrastructure changes, unencrypted HTTP calls may not work as expected, as the typical reason an API call may fail is that HTTP is being redirected to HTTPS.  Please note that any issues that may arise from an unencrypted HTTP call may be resolved by updating to HTTPS-only API calls.   We apologize for any inconvenience this may cause and we look forward to continuing a partnership with you.

For best performance, please review your code to ensure only HTTPS calls are being made to the Zenolio API.

Thanks,
Chris Gauthier
Sr. Manager, Data Center and IT Operations
Zenfolio, Inc.

**Comment by p0las** [comment]
I tried forcing a secure connection for all requests but that didn't seem to change anything. I'm still getting 403
```
 $f = new phpZenfolio("AppName={$appname}");
 $f->setSecureOnly();
```
any help would be greatly appreciated.

**Comment by chris-zenfolio** [comment]
@p0las :  I just looked at the plugin source and it appears the constructor creates https connections by default.  I'm not sure you need to use the setSecureOnly() method.  Please know that I am not a developer and that my findings may not be 100% accurate-YMMV.  If you are able to capture the headers using a cURL command with some output (please be sure to sanitize it before posting publicly!), that would help, especially if you have a RayID header.  :)

**Comment by lildude** [comment]
@p0las Your issue is definitely not due to HTTP vs HTTPS as as @chris-zenfolio has pointed out, the code defaults to HTTPS. It also wouldn't produce a 403 error.

Your 403 indicates a permissions issue. Either you're not authenticating or you don't have the correct access. The latter is quite a common issue with partner accounts which haven't been granted all the necessary access for private albums.

In short, check that you are authenticating correctly and that you have the correct access rights for the account/albums.

**Comment by chris-zenfolio** [comment]
@lildude You're welcome to close out this issue if you are already making all the calls using https by default.  I would recommend adjusting the constructor to not allow unencrypted http links, if possible.  Thank you for the awesome plugin!  If you run into issues, please email our support team.  They can get you in direct contact with myself or whomever is the best resource for your need.

**Comment by p0las** [comment]
@chris-zenfolio my problem is that recently all my websites that rely on pulling data from zenfolio broke. At the same time the lightroom plugin that connects to zenfolio broke too. As far as I know this was caused by zenfolio moving to https but I cannot be sure.
The latest LR plugin seems to be fixed and I was looking for a way to fix phpZenfolio too. So far I have no idea why it is not working. Since the LR plugin works I know that my API access is ok.
@lildude unfortunately I don't know how to check that. I know PHP but I know nothing about cURL to debug the connection.

**Comment by p0las** [comment]
@lildude I just looked at my (very old) code and it appears that I'm using phpZenfolio 1.3
I will update to the latest 2.0.1 and hopefully, this will do the trick. 2.0 seems very different to 1.3 on a quick glance.

**Comment by p0las** [comment]
I have updated to 2.0.1 (installed composer and used it to get phpZenfolio and dependency)
changed the code to:

```
  $f = new phpZenfolio\Client($appname);
  $f->login('olo@polas.net', password);
```

I'm getting:
Client error: `POST https://api.zenfolio.com/api/1.8/zfapi.asmx` resulted in a `403 Forbidden` response: error code: 1010 (Error Code: 403)

I don't think there is much I can change here. my credentials are good (the same ones I log into zenfolio web page), the API is enabled for my user etc.

**Comment by p0las** [comment]
so after a bit more tests I found the problem. For some reason my app name is no longer compatible with zenfolio API. after changing it I got a successful connection. 
All good now.
you can close the ticket. cheers.

**Comment by chris-zenfolio** [comment]
Here is the latest example of 301 redirects:

$ curl -I -X GET "http://api.zenfolio.com"
HTTP/1.1 301 Moved Permanently
Date: Thu, 20 May 2021 18:58:48 GMT
Content-Length: 0
Connection: keep-alive
Cache-Control: no-cache, no-store, must-revalidate
Pragma: no-cache
Expires: -1
Location: http://www.zenfolio.com/
CF-Cache-Status: DYNAMIC
cf-request-id: 0a2cbe12e70000fd9957a74000000001
Set-Cookie:
__cf_bm=fc2a407b9f452d2e2e2c50bf79d362608b7c713d-1621537128-1800-Ad/HWgoQKMOyfVADM2JpQOfn+QnQbFRfALzAjgdBD35FxF6znxfWUFfauR1z2SUsjc2CVl5GHsAGyMLRcEF9/mY=;
path=/; expires=Thu, 20-May-21 19:28:48 GMT; domain=.zenfolio.com;
HttpOnly; SameSite=None
X-Content-Type-Options: nosniff
Server: cloudflare
CF-RAY: 6527cc64ad83fd99-PDX



*Chris Gauthier*

Senior Manager, Data Center and IT Operations

Zenfolio, Inc.

***@***.***



“To invent an airplane is nothing. To build one is something. To fly is
everything.”

          – Ferdinand Ferber,  c. 1898
